main/lsp/lsp_helpers.cc

```diff
@@ -113,6 +114,11 @@ int cmpLocations(const Location &a, const Location &b) {
     return cmpPositions(*a.range->end, *b.range->end);
 }
 
+bool cmpRanges(const Range &a, const Range &b) {
```

Comment:
I wasn't sure what use a proper comparison function for ranges would have so I went with this simple approach. However, since your implementation is pretty clean, it sounds like it can't hurt.